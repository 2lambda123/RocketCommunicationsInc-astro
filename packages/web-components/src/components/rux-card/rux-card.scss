:host {
    // Internal private design tokens
    --card-header-padding: calc(var(--spacing-2) + 2px); //10px
    --card-footer-padding: calc(var(--spacing-2) + 2px); //10px

    /**
     * @prop --body-padding: the card's body padding
     */
    --body-padding: var(--card-footer-padding, calc(var(--spacing-2) + 2px));
    display: block;
}

.hidden,
:host([hidden]) {
    display: none;
}

.rux-card {
    border: 1px solid var(--card-color-border);
    border-radius: var(--radius-base);
    background: var(--color-background-base-default);
    box-shadow: var(--shadow-overlay);
}

.rux-card__header {
    padding-top: calc(var(--card-header-padding) + 4px);
    padding-bottom: var(--card-header-padding);
    padding-left: var(--card-header-padding);
    padding-right: var(--card-header-padding);
    font-family: var(--font-heading-3-font-family);
    font-size: var(--font-heading-3-font-size);
    font-weight: var(--font-heading-3-font-weight);
    letter-spacing: var(--font-heading-3-letter-spacing);
    line-height: var(--font-heading-3-line-height);
    box-sizing: border-box;
    min-height: calc(var(--spacing-12)-2px);
}

.rux-card__body {
    padding: var(--body-padding);
    box-sizing: border-box;
    min-height: 114px;
}

.rux-card__footer {
    padding-left: var(--card-footer-padding);
    padding-right: var(--card-footer-padding);
    padding-top: var(--card-footer-padding);
    padding-bottom: calc(var(--card-footer-padding) + 2px);
    box-sizing: border-box;
    min-height: calc(var(--spacing-12)-2px);
}
