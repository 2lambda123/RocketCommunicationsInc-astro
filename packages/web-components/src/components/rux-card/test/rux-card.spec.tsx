import { newSpecPage } from '@stencil/core/testing'
import { RuxCard } from '../rux-card'

describe('rux-card', () => {
    it('renders', async () => {
        const page = await newSpecPage({
            components: [RuxCard],
            html: `<rux-card></rux-card>`,
        })
        expect(page.root).toMatchSnapshot()
    })
})
