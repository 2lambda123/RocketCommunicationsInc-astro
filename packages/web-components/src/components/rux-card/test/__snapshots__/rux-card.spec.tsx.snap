// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rux-card renders 1`] = `
<rux-card>
  <mock:shadow-root>
    <div class="rux-card">
      <div class="hidden rux-card__header" part="header">
        <slot name="header"></slot>
      </div>
      <div class="rux-card__body" part="body">
        <slot></slot>
      </div>
      <div class="hidden rux-card__footer" part="footer">
        <slot name="footer"></slot>
      </div>
    </div>
  </mock:shadow-root>
</rux-card>
`;
